apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.tsx

```diff
@@ -0,0 +1,44 @@
+import type { IRoom } from '@rocket.chat/core-typings';
+import { useMutableCallback } from '@rocket.chat/fuselage-hooks';
+import { useUserRoom } from '@rocket.chat/ui-contexts';
+import React from 'react';
+
+import { useTabBarClose } from '../../../contexts/ToolboxContext';
+import RoomInfo from './RoomInfo';
+import { useCanEditRoom } from './hooks/useCanEditRoom';
+
+type RoomInfoWithDataProps = {
+	rid: IRoom['_id'];
+	openEditing?: () => void;
+	onClickBack: () => void;
+	onEnterRoom: (room: IRoom) => void;
+	resetState: () => void;
+};
+
+const RoomInfoWithData = ({ rid, openEditing, onClickBack, onEnterRoom, resetState }: RoomInfoWithDataProps) => {
+	const onClickClose = useTabBarClose();
+	const room = useUserRoom(rid);
```

Comment:
It would be neat to use `useRoom` here instead of `useUserRoom`, as the data comes from context.